### PR TITLE
[SYCL] Queue has to be released later than event

### PR DESCRIPTION
**Pull request by againull**

### CODE REVIEW

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -6028,20 +6028,22 @@ static pi_result piEventReleaseInternal(pi_event Event) {
...
-  // We intentionally incremented the reference counter when an event is
-  // created so that we can avoid pi_queue is released before the associated
-  // pi_event is released. Here we have to decrement it so pi_queue
-  // can be released successfully.
-  if (Event->Queue) {
-    PI_CALL(piQueueReleaseInternal(Event->Queue));
-  }
-
+  // Save pointer to the queue before deleting/resetting event.
+  auto Queue = Event->Queue;
```
Comment (smaslov-intel) [inline sycl/plugins/level_zero/pi_level_zero.cpp:6032]: could explain what goes wrong if we first release queue and then (immediately after) release the event?

Comment (againull) [reply sycl/plugins/level_zero/pi_level_zero.cpp:6032]: When we add an event to the cache we need to check whether profiling is enabled or not, so we access properties of the queue (it's an invalid access of we release queue before that).

Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:6032]: I think it's worth adding a source comment about that

Comment (againull) [reply sycl/plugins/level_zero/pi_level_zero.cpp:6035]: Ok, added a comment

**Review by smaslov-intel — APPROVED** [review]
Thanks

**Pull request merged**